File: Difficulty: Easy/Remove Duplicates/remove-duplicates.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:

    string removeDups(string s) {
        int n = s.size();
        // frequency make its size 255 as per ASCII values
        vector<int> ch(255, 0);
    
        // Create result string
        string ans = "";
    
        for (int i = 0; i < n; i++) {
            // Check if current character's frequency is 0 or
            // not
            if (ch[s[i]] == 0) {
    
                // Add char if frequency is 0
                ans.push_back(s[i]);
    
                // Increment frequency by 1
                ch[s[i]]++;
            }
        }
        return ans;
    }
};

//{ Driver Code Starts.

int main() {

    ios_base::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;

        Solution ob;
        cout << ob.removeDups(s) << "\n";
    }

    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Remove Duplicates/remove-duplicates.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String s = read.readLine();

            Solution ob = new Solution();
            String result = ob.removeDups(s);

            System.out.println(result);
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    String removeDups(String str) {
        HashSet<Integer> arr = new HashSet<>();
        StringBuilder S = new StringBuilder();
        for(int i = 0;i < str.length(); i++)
        {
            int num = str.charAt(i);   
            if(arr.contains(num))
            {
                continue;
            }
            else
            {
                arr.add(num);
                S.append((char)str.charAt(i));
            }
        }
        return S.toString();
    }
}